<div id="navegacion" prev="../Tema_4/t4_funciones_02_pdi.html" 
                     next="../Ejercicios_Tema_3/ej2.html"></div>

<div class="section-title">Sección 2.8</div>

# Ejercicios

## Ejercicio 1

Escribe el código Python que representa la siguiente expresión:

$$\frac{3}{4} + 2 * ( 5^4 - \frac{1}{2} )$$



In [7]:
# Sol:
print( 3/4+2*(pow(5,4)-1/2))



resultado 1249.75


Escribe una función en Python llamada __valor_expresion__ que calcule el valor de la expresión:

$$\frac{3}{4} + 2 * ( x^4 - \frac{1}{2} )$$

para un cierto valor de $x$. La función tiene un único argumento posicional de tipo numérico y devuelve el resultado de aplicar la expresión anterior.

In [16]:
# Sol
def valor_expresion (x):
    return 3/4+2*(pow(x,4)-1/2)



resultado 1249.75


Ejecuta la función __valor_expresion__ para los valores 5 y 3. 

__Nota:__ 

* `valor_expresion(5)` devuelve como resultado `1249.75`
* `valor_expresion(3)` devuelve como resultado `161.75`

In [18]:
# Sol
print(valor_expresion(5))
print(valor_expresion(3))

resultado 1249.75


## Ejercicio 2

Define una función llamada __raiz_cubica__ que devuelva el valor de $\sqrt[3]{x}$. Recuerda que  $\sqrt[3]{x}$ = ${x}^{\frac{1}{3}}$


__Nota:__ 

* `raiz_cubica(5)` devuelve como resultado `1.7099759466766968`

In [22]:
# Sol:
def raiz_cubica(x):
    return pow(x, 1/3)

print(raiz_cubica(5))

1.7099759466766968


## Ejercicio 3

Escribe una función llamada __pertenencia__ que reciba dos parámetros de entrada. El primer parámetro es una lista de elementos y el segundo es un elemento. La función devuelve el valor `True` si el elemento se encuentra en la lista. En otro caso devuelve `False`. 

* Puedes usar la instrucción `if-else` 
* Puedes usar la función `in` para saber si un elemento está en una secuencia

__Nota:__

* `pertenencia([1, 3, 6], 1)` devuelve el valor `True`
* `pertenencia([1, 3, 6], 5)` devuelve el valor `False`
* `pertenencia([1,'Hi', 6, 'Hola'] , 'Hola')` devuelve el valor `True`

In [45]:
# Sol:
def pertenencia(array, valor):
    return valor in array

print(pertenencia([1, 3, 6], 1))
print(pertenencia([1, 3, 6], 5))
print(pertenencia([1,'Hi', 6, 'Hola'] , 'Hola'))

True
False
True


## Ejercicio 4

La función `upper` del tipo string (`str`) genera una nueva cadena de caracteres, donde todos los caracteres son mayúsculas.
Ejecuta la siguente celda para ver el efecto de la función `upper` al aplicarlo sobre la cadena `s`:

In [51]:
s = 'Hola, me llamo Iñigo Montoya, tú mataste a mi padre. ... Prepárate a morir'
s.upper()

'HOLA, ME LLAMO IÑIGO MONTOYA, TÚ MATASTE A MI PADRE. ... PREPÁRATE A MORIR'

Fíjate que `upper` no modifica la cadena (las cadenas son inmutables), sino que devuelve otra cadena diferente.

In [53]:
s    # la cadena s no cambia de valor

'Hola, me llamo Iñigo Montoya, tú mataste a mi padre. ... Prepárate a morir'

Para guardar el resultado de `upper` usamos la asignación:

In [55]:
t = s.upper()

Podemos  preguntar por el contenido de `t`:

In [57]:
t

'HOLA, ME LLAMO IÑIGO MONTOYA, TÚ MATASTE A MI PADRE. ... PREPÁRATE A MORIR'

* Genera una cadena `t`  igual que  `s`  pero en minúsculas (función `lower`). Calcula la longitud de `t`.
* Ejecuta la función  `split(',')`. Fíjate en lo que devuelve esa función. 
* Ejecuta la función  `title()`.
* Ejecuta la función  `replace(' ', '_')`. Fíjate en lo que devuelve esa función.
* Ejecuta `dir(str)` para listar las funciones disponibles para trabajar con cadenas.

In [87]:
# Sol: 
t=s.lower()
len(t)
t.split(',')
t.title()
t.replace(' ', '_')
dir(str)


'hola,_me_llamo_iñigo_montoya,_tú_mataste_a_mi_padre._..._prepárate_a_morir'

--------------------